### PR TITLE
Convert kepler_third_law and freefall_velocity to C

### PR DESCRIPTION
**Pull request by maxpkatz**
## PR checklist

- [x] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Exec/science/wdmerger/wdmerger_util.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    // First, solve for the orbit in the reduced one-body problem, where
+    // an object of mass mu orbits an object with mass M located at r = 0.
+    // For this we follow Carroll and Ostlie, Chapter 2, but many texts discuss this.
+    // Note that we use the convention that phi measures angle from aphelion,
+    // which is opposite to the convention they use.
+
+    if (period > 0.0_rt && a < 0.0_rt)
+    {
+        a = std::pow(C::Gconst * M * period * period / (4.0_rt * M_PI * M_PI), 3); // C + O, Equation 2.37
+    }
```
Comment (zingale) [inline Exec/science/wdmerger/wdmerger_util.cpp:38]: shouldn't this be to the 1/3 power? not 3?

File: Exec/science/wdmerger/wdmerger_util.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    r_2  =  (mu / mass_2) * r;
+
+    v_1r = -(mu / mass_1) * v_r;
+    v_2r =  (mu / mass_2) * v_r;
+
+    v_1p = -(mu / mass_1) * v_phi;
+    v_2p =  (mu / mass_2) * v_phi;
+}
+
+}
```
Comment (zingale) [inline Exec/science/wdmerger/wdmerger_util.cpp:71]: there were some additional bits in the Fortran version that checked to make sure that the domain was big enough to hold the stars.  Did you intend for that to be cut out?

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-11-18-002/index.html

File: Exec/science/wdmerger/wdmerger_util.cpp (hunk 1 above)
Comment (maxpkatz) [reply Exec/science/wdmerger/wdmerger_util.cpp:38]: Good catch, fixed

File: Exec/science/wdmerger/wdmerger_util.cpp (hunk 2 above)
Comment (maxpkatz) [reply Exec/science/wdmerger/wdmerger_util.cpp:71]: I do like that check, however the problem is that it depended on access to `physbc_lo` which is available as a 'global' in Fortran but not currently in C++. So I punted on that for now, and moved that check to the calling routine of kepler_third_law in Fortran.

**Review by zingale — APPROVED** [review]

**Pull request merged**